components/feature/addons/src/main/java/mozilla/components/feature/addons/migration/SupportedAddonsChecker.kt

```diff
@@ -0,0 +1,204 @@
+ * @property applicationContext The application context.
+ * @param frequency (Optional) indicates how often checks should be performed, defaults
+ * to once per day.
+ * @param onNotificationClickIntent (Optional) Indicates which intent should be passed to the notification
+ * when new supported add-ons are available.
+ */
+@Suppress("TooManyFunctions", "LargeClass")
+class DefaultSupportedAddonsChecker(
+    private val applicationContext: Context,
+    private val frequency: Frequency = Frequency(1, TimeUnit.DAYS),
+    onNotificationClickIntent: Intent = createDefaultNotificationIntent(applicationContext)
+) : SupportedAddonsChecker {
+    private val logger = Logger("DefaultSupportedAddonsChecker")
+
+    init {
+        SupportedAddonsWorker.onNotificationClickIntent = onNotificationClickIntent
+    }
+
+    /**
+     * See [SupportedAddonsChecker.registerForChecks]
+     */
+    override fun registerForChecks() {
+        WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                CHECKER_UNIQUE_PERIODIC_WORK_NAME,
+                ExistingPeriodicWorkPolicy.KEEP,
+                createPeriodicWorkerRequest()
+        )
+        logger.info("Register check for new supported add-ons")
+    }
+
+    /**
+     * See [SupportedAddonsChecker.unregisterForChecks]
+     */
+    override fun unregisterForChecks() {
+        WorkManager.getInstance(applicationContext)
+                .cancelUniqueWork(CHECKER_UNIQUE_PERIODIC_WORK_NAME)
+        logger.info("Unregister check for new supported add-ons")
+    }
+
+    @VisibleForTesting
+    internal fun createPeriodicWorkerRequest(): PeriodicWorkRequest {
+        return PeriodicWorkRequestBuilder<SupportedAddonsWorker>(
+                frequency.repeatInterval,
+                frequency.repeatIntervalTimeUnit
+        ).apply {
+            setConstraints(getWorkerConstrains())
+            addTag(WORK_TAG_PERIODIC)
+        }.build()
+    }
+
+    @VisibleForTesting
+    internal fun getWorkerConstrains() = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+
+    companion object {
+        private const val IDENTIFIER_PREFIX = "mozilla.components.feature.addons.migration"
+
+        @VisibleForTesting
+        internal const val CHECKER_UNIQUE_PERIODIC_WORK_NAME =
+                "$IDENTIFIER_PREFIX.DefaultSupportedAddonsChecker.periodicWork"
+
+        /**
+         * Identifies all the workers that periodically check for new add-ons.
+         */
+        @VisibleForTesting
+        internal const val WORK_TAG_PERIODIC =
+                "$IDENTIFIER_PREFIX.DefaultSupportedAddonsChecker.periodicWork"
+
+        internal fun createDefaultNotificationIntent(content: Context): Intent {
+            return content.packageManager.getLaunchIntentForPackage(content.packageName)?.apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK
```

Comment:
Perhaps `FLAG_ACTIVITY_NEW_TASK or FLAG_ACTIVITY_CLEAR_TASK`? I could be wrong here but we don't want multiple intent launch instances.

Reply:
Thanks for caching this :)